sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -763,19 +781,22 @@ public ServiceBusReceiverAsyncClient buildAsyncClient() {
             final ReceiverOptions receiverOptions = new ReceiverOptions(receiveMode, prefetchCount,
                 sessionId, isRollingSessionReceiver(), maxConcurrentSessions);
 
+            final UnnamedSessionManager sessionManager;
+
             if (CoreUtils.isNullOrEmpty(sessionId)) {
-                final UnnamedSessionManager sessionManager = new UnnamedSessionManager(entityPath, entityType,
+                sessionManager = new UnnamedSessionManager(entityPath, entityType,
                     connectionProcessor, connectionProcessor.getRetryOptions().getTryTimeout(), tracerProvider,
```

Comment:
Not sure why we are passing connectionProcessor and connectionProcessor.getRetryOptions().getTryTimeout(). If the session manager has the connection processor, it should be able to get the retry options from it anyway. Also, it's weird to see retry options in connection processor.

Reply:
1. I removed `connectionProcessor.getRetryOptions().getTryTimeout()` . 
2. `retry options in connection processor` This is being passed on to AMQP Channel Processor in  core-amqp library for retry policy.